hydra-filters/src/main/java/com/addthis/hydra/data/filter/bundle/BundleFilterFirstValue.java

```diff
@@ -104,10 +77,10 @@ public boolean filterExec(Bundle bundle) {
                     ValueArray arr = v.asArray();
                     if (arr.size() > 0) {
                         if (which != null) {
-                            ValueObject fieldName = ValueFactory.create(bound[i].getName());
-                            bundle.setValue(bound[end + 1], fieldName);
+                            ValueObject fieldName = ValueFactory.create(((CachingField) in[i]).name);
```

Comment:
you can probably declare the field directly as a `CachingField` if that is going to be enforced later anyway